sql/wsrep_utils.cc

```diff
@@ -331,13 +331,20 @@ static int execvpe(const char *file, char *const argv[], char *const envp[]) {
 #define STDIN_FD 0
 #define STDOUT_FD 1
 
-process::process(const char *cmd, const char *type, char **env)
-    : str_(cmd ? strdup(cmd) : strdup("")), io_(NULL), err_(0), pid_(0) {
+process::process(const char *cmd, const char *type, char **env, bool execute_immediately)
+    : str_(cmd ? strdup(cmd) : strdup("")), io_(NULL), io_w_(NULL), err_(0), pid_(0)
+  {
+    if (execute_immediately)
```

Comment:
what is the significance of this execute immediately? Can you add a comment?

Reply:
The original code would execute the cmd as part of the constructor.  I moved the execution outside of the constructor (so that we could skip execution if an error occurred).  To avoid changing existing code, I left the ability to execute the code immediately as part of the constructor.  (will add a comment)

Reply:
comment added to wsrep_utils.h